contracts/identity/Identity.sol

```diff
@@ -72,7 +73,7 @@ contract Identity is ProxyStorage {
         require(validateTimeLimit(timeLimit), "The transaction expired");
         require(validateSignature(hash, signature), "The transaction signature is not valid");
 
-        if (nonce == 0) {
+        if (nonce == 0 || nonce >= maxNonce) {
```

Comment:
Maybe add a comment that this allows to use a random nonce >= maxNonce to allow to replay similar transactions.